### ISSUE TITLE
Separate read-classification from report generation

### ISSUE THREAD

**Issue by maubarsom** [infphilo/centrifuge#111]
Hi,

This suggestion is inspired from the centrifuge manual, where it states:

> As the GenBank database is incomplete (i.e., many more genomes remain to be identified and added), and reads have sequencing errors, classification programs including Centrifuge often report many false assignments. In order to perform more conservative analyses, users may want to discard assignments for reads having a matching length (8th column in the output of Centrifuge) of 40% or lower. It may be also helpful to use a score (4th column) for filtering out some assignments. Our future research plans include working on developing methods that estimate confidence scores for assignments.

If I was to follow this suggestion, I am able to filter the assignments out of the Centrifuge output, but I could not find a script to regenerate a report for the filtered sample. I looked around and it seems there is a centrifuge_report.cpp with a main() function, but it appears the make file does not generate a binary for it. 

Thanks!

**Comment by mourisl** [comment]
The filter can be easily done with "awk" or your own script to process the classification output of Centirfuge.

**Comment by khyox** [comment]
@maubarsom, perhaps [Recentrifuge](http://www.recentrifuge.org) could help you with that.

**Comment by mourisl** [comment]
That's a good suggestion to compute the report file just based on classification output file. We'll implement this in future.

For now, you can convert the output to kraken style report with script centrifuge-kreport, and there is a column for the proportion of reads belong to that taxonomy subtree. This number, though just from directly counting and not based on EM algorithm, could be a good approximation for abundance.

**Comment by anbjork** [comment]
Hi Centrifuge devs and OP,
Yes, please implement that feature! That would be gold for me as I would like to do the exact same thing as OP. Thank you for the advice to use centrifuge-kreport. It seems like a bit of a shame to lose out on the EM estimation of abundances.

There is an option 
--min-totallen <int>
described in the information displayed upon calling
centrifuge -h
The decription reads "minimum summed length of partial hits per read (default 0)".
It sound like exactly what I and OP want (except Centrifuge has to be rerun), but when trying it out the option seems to be ignored by Centrifuge. Is it not implemented any more/yet?

Have a nice day!

**Comment by depancao** [comment]
Yes, please implement that feature! I can't wait for another year!
